main.go

```diff
@@ -232,6 +232,27 @@ func main() {
 				&corev1.Secret{}: {
 					Field: fields.SelectorFromSet(fields.Set{"metadata.name": propagatorctrl.EncryptionKeySecret}),
 				},
+				&clusterv1.ManagedCluster{}: {
+					Transform: func(obj interface{}) (interface{}, error) {
+						cluster := obj.(*clusterv1.ManagedCluster)
+						// All that ManagedCluster objects are used for is to check their existence to see if a
+						// namespace is a cluster namespace.
+						guttedCluster := &clusterv1.ManagedCluster{}
+						guttedCluster.SetName(cluster.Name)
```

Comment:
Label not used? I found this `ManagedClusterLabels map[string]string`

Reply:
It's a bit confusing but that ManagedCluster isn't queried by the controller-runtime caching client but it's coming from the policy templating client (which uses the caching dependency watcher library):
https://github.com/open-cluster-management-io/governance-policy-propagator/blob/a06836bba155f9ca30edd0ded268d38d222baf3f/controllers/propagator/propagation.go#L285